src/repl.ts

```diff
@@ -196,7 +210,7 @@ async function startREPL(preserveFocus: boolean) {
                     shellPath: exepath,
                     shellArgs: jlarg1.concat(jlarg2),
                     env: {
-                        JULIA_EDITOR: process.platform == 'darwin' ? `"${path.join(vscode.env.appRoot, 'bin', 'code')}"` : `"${process.execPath}"`,
+                        JULIA_EDITOR: process.platform == 'darwin' ? `"${path.join(vscode.env.appRoot, 'bin', 'code')}"` : get_editor(),
```

Comment:
Actually, while you are editing this part: could you move this check for MacOS into the `get_editor` function as well? Then it would apply for both cases where we set `JULIA_EDITOR`, which it should, i.e. you could fix an old bug with this PR as well :)

Reply:
done.